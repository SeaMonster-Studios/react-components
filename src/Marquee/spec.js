import React from "react"
import { createSerializer } from "jest-emotion"
import * as emotion from "emotion"
//
import { render, cleanup } from "react-testing-library"
import { Marquee } from "./"

afterEach(cleanup)

expect.addSnapshotSerializer(createSerializer(emotion))

describe("Marquee", () => {
  it("Renders", () => {
    const { getByTestId } = renderSetup()
    const Marquee = getByTestId("component-content-marquee")

    expect(Marquee).toBeDefined()
    expect(Marquee).toMatchSnapshot()
  })
})

function renderSetup(overrides) {
  const props = {
    image: "https://picsum.photos/1200/500",
    styles: ``,
    ...overrides,
  }

  const wrapper = render(
    <Marquee {...props}>
      <h2>
        Try our newest flavor â€” <br />
        <em>banana by bananarama</em>
      </h2>
      <button>Get some</button>
    </Marquee>,
  )

  return {
    ...wrapper,
    props,
  }
}
