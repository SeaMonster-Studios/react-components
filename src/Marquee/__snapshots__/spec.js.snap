// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Marquee Renders 1`] = `
.emotion-0 {
  background: linear-gradient(rgba(255,255,255,0),rgba(255,255,255,0)),url(https://picsum.photos/1200/500);
  background-size: cover;
  background-position: center center;
}

<div
  class="emotion-0 emotion-1"
  data-testid="component-content-marquee"
>
  <h2>
    Try our newest flavor â€” 
    <br />
    <em>
      banana by bananarama
    </em>
  </h2>
  <button>
    Get some
  </button>
</div>
`;
